version-catalog/api/src/main/kotlin/io/spine/internal/catalog/CatalogRecords.kt

```diff
@@ -0,0 +1,153 @@
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.internal.catalog
+
+import org.gradle.api.initialization.dsl.VersionCatalogBuilder
+
+/**
+ * A single, atomic unit, which can be written into a version catalog.
+ *
+ * Records say what information is necessary to write one or another item into
+ * a version catalog. Theirs properties strictly reflect the way, an item is
+ * stored in the catalog.
+ *
+ * Similarly to [notations][CatalogNotation], records define what information is
+ * needed to create one or another catalog-compatible item. But in contrast to
+ * notations, records are citizens of an implementation site. They sas what is needed
+ * from the point of view of how this information is actually stored in the catalog.
+ *
+ * Consider the following example:
+ *
+ * ```
+ * // A user defines a plugin, using the corresponding notation.
+ *
+ * val plugin = object : PluginNotation {
+ *     override val id: String = "kotlin-jvm"
+ *     override val module: String = "org.jetbrains:kotlin-jvm"
+ *     override val version: String = "1.6.21"
+ *     override val alias: Alias = "kotlin.jvm"
+ * }
+ * ```
+ *
+ * In order to represent this notation in [VersionCatalogBuilder], three records
+ * should be written there: [VersionRecord], [LibraryRecord], [PluginRecord].
+ *
+ * This interface, in particular, doesn't describe any concrete catalog item.
+ * It just serves as a common foundation for other records.
+ */
+interface CatalogRecord {
+
+    /**
+     * A name, by which this record will be known in the catalog.
+     *
+     * For example: `kotlin.stdLib.common.jvm`.
+     */
+    val alias: Alias
+
+    /**
+     * Writes this record to the given catalog.
+     */
+    fun writeTo(catalog: VersionCatalogBuilder)
+}
+
+/**
+ * Represents a version item, which can be directly written into a version catalog.
+ *
+ * [value] is a string representation of a version.
+ *
+ * For example: `2.0.0-SNAPSHOT.21`.
+ */
+internal data class VersionRecord(
+    override val alias: Alias,
+    val value: String
+) : CatalogRecord {
+
+    override fun writeTo(catalog: VersionCatalogBuilder) {
+        catalog.version(alias, value)
+    }
+}
+
+/**
+ * Represents a library item, which can be directly written into a version catalog.
+ *
+ * [module] is a group and artifact of a library, seperated by a colon.
+ *
+ * For example: `io.spine:spine-core`.
+ *
+ * Version for the library is obtained by the reference. Thus, the given [versionRef]
+ * should point to a [VersionRecord].
+ */
+internal data class LibraryRecord(
+    override val alias: Alias,
+    val module: String,
+    val versionRef: Alias
```

Comment:
I also don't understand why this isn't a `VersionRecord`, or something more strongly typed than just a `String`.